spacelift/resource_worker_pool.go

```diff
@@ -209,6 +215,32 @@ func resourceWorkerPoolRead(ctx context.Context, d *schema.ResourceData, meta in
 func resourceWorkerPoolUpdate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
 	name := d.Get("name").(string)
 
+	// If CSR has changed, use workerPoolReset mutation
+	if d.HasChange("csr") {
+		var resetMutation struct {
+			WorkerPool structs.WorkerPool `graphql:"workerPoolReset(id: $id, certificateSigningRequest: $csr)"`
+		}
+
+		csrString := d.Get("csr").(string)
+		resetVariables := map[string]interface{}{
+			"id":  toID(d.Id()),
+			"csr": graphql.String(csrString),
+		}
+
+		if err := meta.(*internal.Client).Mutate(ctx, "WorkerPoolReset", &resetMutation, resetVariables); err != nil {
+			return diag.Errorf("could not reset worker pool token: %v", internal.FromSpaceliftError(err))
+		}
+
+		// Update the config with the new token info
+		d.Set("config", resetMutation.WorkerPool.Config)
+
+		// Return early if we only had a CSR change
+		if !d.HasChanges("name", "description", "labels", "space_id") {
```

Comment:
Why do we need this?

Can we achieve the same behaviour without specifying the list of fields? I worry this will get out of sync in the future.

Reply:
It should work without it, but with this simple if we are avoiding 2 unnecessary API calls so I believe we should keep it. However I agree that listing all fields here is bad and error prone. I replaced it with `if !d.HasChangeExcept("csr") {`